packages/react-instantsearch-core/src/lib/wrapPromiseWithState.ts

```diff
@@ -0,0 +1,60 @@
+// Taken from https://github.com/apollographql/apollo-client/blob/ba3e7d9fa7d46e4c636148bbf01552833db0ceda/src/utilities/promises/decoration.ts
```

Comment:
cc @mattcarrollcode, it would be nice if React exposes this so it doesn't need to be reimplemented :)